source/slang/lower-to-ir.cpp

```diff
@@ -2307,7 +2312,8 @@ struct DeclLoweringVisitor : DeclVisitor<DeclLoweringVisitor, LoweredValInfo>
 
         // For now, we don't have an IR-level representation
         // for the type itself.
-        return LoweredValInfo();
+        return LoweredValInfo::simple(context->irBuilder->getTypeVal(DeclRefType::Create(context->getSession(), 
+            DeclRef<Decl>(decl, nullptr))));
```

Comment:
I'm wondering why this case came up... The reason that type declarations previously didn't have IR values registered for them during lowering was that they'd only appear in types anyway, and so they weren't needed at the value level. What changed?

Reply:
This shows up because a witness table now needs to store entries for associated types, whose values are currently decls to the actual type definition itself. For example, a `struct S : IBase` may have a `typedef` member to statisfy an associatedtype requirement. There will be an witness table entry pointing to the typedef Decl in `S` serving as the conformance witness. This leads to such visiting functions being invoked when translating the witness table into an IRValue. However the current implementation is silppery because we should really store a fully resolved Type in the witness table entry rather than having to reconstruct a DeclRefType here (which loses the necessary substitutions if the `S` is a generic or defined within a generic parent).